tests/cisco_ios/show_crypto_ipsec_sa_detail/cisco_ios_show_crypto_ipsec_sa_detail.yml

```diff
@@ -0,0 +1,286 @@
+---
+parsed_sample:
+  - interface: "Tunnel1"
+    crypto_map_tag: "Tunnel1-head-0"
+    local_addr: "1.2.3.4"
+    protected_vrf: "(none)"
+    local_ident_addr: "1.2.3.4"
+    local_ident_mask: "255.255.255.255"
+    local_ident_prot: "47"
+    local_ident_port: "0"
+    remote_ident_addr: "2.2.2.2"
+    remote_ident_mask: "255.255.255.255"
+    remote_ident_prot: "47"
+    remote_ident_port: "0"
+    current_peer: "2.2.2.2"
+    port: "4500"
+    action: "PERMIT"
+    flags: "origin_is_acl,"
+    pkts_encaps: "4981028"
+    pkts_encrypt: "4981028"
+    pkts_digest: "4981028"
+    pkts_decaps: "4112421"
+    pkts_decrypt: "4112421"
+    pkts_verify: "4112421"
+    pkts_compressed: "0"
+    pkts_decompressed: "0"
+    pkts_not_compressed: "0"
+    pkts_compress_failed: "0"
+    pkts_not_decompressed: "0"
+    pkts_decompress_failed: "0"
+    pkts_no_sa: "0"
+    pkts_invalid_sa: "0"
+    pkts_encaps_failed: "0"
+    pkts_decaps_failed: "0"
+    pkts_invalid_prot: "0"
+    pkts_verify_failed: "0"
+    pkts_invalid_identity: "0"
+    pkts_invalid_len: "0"
+    pkts_replay_rollover_send: "0"
+    pkts_replay_rollover_recv: "0"
+    pkts_replay_failed: "0"
+    pkts_tagged: "0"
+    pkts_untagged: "0"
+    pkts_not_tagged: "0"
+    pkts_not_untagged: "0"
+    pkts_internal_err_send: "0"
+    pkts_internal_err_recv: "0"
+    local_crypto_endpt: "1.2.3.4"
+    remote_crypto_endpt: "2.2.2.2"
+    plaintext_mtu: "1442"
+    path_mtu: "1500"
+    ip_mtu: "1500"
+    ip_mtu_idb: "GigabitEthernet8"
+    current_outbound_spi_hex: "0x1234ABCD"
+    current_outbound_spi_dec: "305441741"
+    pfs: "N"
+    dh_group: "none"
+    sa_orientation: "inbound"
+    sa_type: "esp"
+    sa_spi_hex: "0xABCD1234"
+    sa_spi_dec: "2882343476"
+    sa_transform: "esp-256-aes esp-sha-hmac"
+    sa_in_use_settings: "Transport UDP-Encaps"
+    sa_conn_id: "124"
+    sa_flow_id: "Onboard VPN:124"
+    sa_sibling_flags: "80000000"
+    sa_crypto_map: "Tunnel1-head-0"
+    sa_lifetime_kbytes: "4332650"
```

Comment:
Instead of creating duplicate records - one for inbound and one for outbound - I think it would be wise to mimic what the Cisco ASA template does for this command and combines them with different key names. For example:
https://github.com/networktocode/ntc-templates/blob/5523fb9ddd3c0baee04858a6fc7ad2bd90988bda/tests/cisco_asa/show_crypto_ipsec_sa/cisco_asa_show_crypto_ipsec_sa.yml#L48-L71

Reply:
The reason I opted to make individual records for each SA is because it is possible for multiple SAs to exist inbound and outbound.
EG: https://community.cisco.com/t5/vpn/multiple-inbound-and-outbound-sas-being-established/td-p/2730901
I'm also worried about multiple SAs during a rekey, but haven't been able to find a good example of that.

Reply:
In that case can you please update your test files to include an example of this?

Reply:
Added another set of tests.